### PR TITLE
[DotNetCore] Add support for new templates location in 3.0 preview8

### PR DESCRIPTION
**Pull request by rodrmoya**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/953251

### CODE REVIEW

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs [hunk 1]
```diff
@@ -198,6 +199,37 @@ public static int Compare (DotNetCoreVersion x, DotNetCoreVersion y)
 			return Compare (x, y) >= 0;
 		}
 
+		public string GetTemplatesDirectory (Func<string, bool> directoryExists, Func<string, IEnumerable<string>> enumerateDirectories)
+		{
```
Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Still do not think we should have this method in the DotNetCoreVersion class. I think it should be a separate class.

Comment (rodrmoya) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Ok, will move it back to where it was

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs [hunk 2]
```diff
@@ -198,6 +199,37 @@ public static int Compare (DotNetCoreVersion x, DotNetCoreVersion y)
...
+						.Where (v => v.Major < Major || (v.Major == Major && v.Minor <= Minor))
+						.OrderByDescending (v => v)
+						.Select (v => availableTemplates [v])
+						.FirstOrDefault ();
+				}
+			}
+
+			return Directory.Exists (templatesDirectory) ? templatesDirectory : string.Empty;
+		}
+
```
[the remark this answers is not in the document]
Comment (rodrmoya) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Yes, see the 2nd commit with the tests, they test this exactly, or do you mean something else?

[the remark this answers is not in the document]
Comment (mrward) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Sorry I think this got re-added, even though it was there when I reviewed originally a while ago :)

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs (hunk 1 above)
Comment (rodrmoya) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Done

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs [hunk 3]
```diff
@@ -198,6 +199,37 @@ public static int Compare (DotNetCoreVersion x, DotNetCoreVersion y)
 			return Compare (x, y) >= 0;
 		}
 
+		public string GetTemplatesDirectory ()
```
[the remark this answers is not in the document]
Comment (rodrmoya) [reply main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: Done

File: main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs [hunk 4]
```diff
@@ -26,6 +26,7 @@
 
 using System;
 using System.IO;
+using System.Collections.Generic;
 using System.Linq;
```
Comment (mrward) [inline main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreVersion.cs]: I guess this can be removed - no harm leaving it though 😄

**Pull request merged**